bigquery/tests/unit/test_job.py

```diff
@@ -4168,6 +4168,22 @@ def test_result_w_timeout(self):
         self.assertEqual(query_request[1]["query_params"]["timeoutMs"], 900)
         self.assertEqual(reload_request[1]["method"], "GET")
 
+    def test_result_w_page_size(self):
+        query_resource = {
+            "jobComplete": True,
+            "jobReference": {"projectId": self.PROJECT, "jobId": self.JOB_ID},
+            "schema": {"fields": [{"name": "col1", "type": "STRING"}]},
+            "totalRows": "4",
+        }
+        connection = _make_connection(query_resource)
+        client = _make_client(self.PROJECT, connection=connection)
+        resource = self._make_resource(ended=True)
+        job = self._get_target_class().from_api_repr(resource, client)
+
+        result = job.result(page_size=3)
+
+        self.assertEqual(result.total_rows, 4)
```

Comment:
Let's assert that [tabledata.list](https://cloud.google.com/bigquery/docs/reference/rest/v2/tabledata/list) is called with the right page size (`maxResults`) as a parameter.

See:

https://github.com/googleapis/google-cloud-python/blob/097572451bc596ac676d2c16557c79d3944641c4/bigquery/tests/unit/test_client.py#L4197

Specifically, I'd like to see something like:

```
tabledata_path = "/projects/%s/datasets/%s/tables/%s/data" % (
    self.PROJECT,
    self.DS_ID,
    self.TABLE_ID,
)
conn.api_request.assert_called_once_with(
    method="GET", path=tabledata_path, query_params={"maxResults": 3}
)
```

Reply:
Hello Tim,
Thanks for the feedback but I think I need a hand.
Do you mean that the maxResults should be directly linked to the page_size ? The api documentation seems to say it  [link](
https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs/query#queryrequest) but the code don't
https://github.com/googleapis/google-cloud-python/blob/097572451bc596ac676d2c16557c79d3944641c4/bigquery/google/cloud/bigquery/client.py#L1949-L1963

Reply:
Confusingly, `page_size` in Python is used to set `maxResults` in the API request, since that's what BigQuery calls its page size parameter.

I've updated your PR to test for this.

Reply:
Okay I see...
Many thanks for the help on the unit test !